src/GraphQL/GraphQLExtensions.cs

```diff
@@ -502,15 +509,59 @@ public static bool DoTypesOverlap(IGraphType typeA, IGraphType typeB)
 
         private static readonly NullValue _null = new NullValue();
 
+        /// <summary>
+        /// Returns a value indicating whether the provided value is a valid default value
+        /// for the specified input graph type.
+        /// </summary>
+        public static bool IsValidDefault(this IGraphType type, object value)
+        {
+            if (type == null)
+                throw new ArgumentNullException(nameof(type));
+
+            if (type is NonNullGraphType nonNullGraphType)
+            {
+                return value == null ? false : nonNullGraphType.ResolvedType.IsValidDefault(value);
+            }
+
+            if (value == null)
+            {
+                return true;
+            }
+
+            if (type is ListGraphType listGraphType)
+            {
+                if (!(value is IEnumerable list))
+                    return false;
+
+                foreach (var item in list)
+                {
+                    if (!IsValidDefault(listGraphType.ResolvedType, item))
+                        return false;
+                }
+
+                return true;
+            }
+
+            if (type is IInputObjectGraphType inputObjectGraphType)
+            {
+                return IsValidDefault(inputObjectGraphType, value);
```

Comment:
Ooops. StackOverflow here. See `IsValidLiteralValue` - you need to iterate over `inputObjectGraphType.Fields`.

Reply:
Also need to write a test with complex default value.

Reply:
Did you mean `inputObjectGraphType.IsValidDefault(value);` ? (instance method)

The compiler should prefer it over the extension method when called like this, although I would still call it dangerous.

Reply:
```suggestion
                return inputObjectGraphType.IsValidDefault(value);
```

Reply:
Suggestion shown above was committed.  Ok?

Reply:
yes